### PR TITLE
[Apps] Accept model-ids for scaffolding apps 

### PR DESCRIPTION
**Pull request by snoe**
Accept model-ids when scaffolding
- does not create implicit actions for existing models
- does add action buttons for existing actions on the selected models (`requires_pk` determines if an action goes on the list or detail page)

### CODE REVIEW

File: src/metabase/api/app.clj [hunk 1]
```diff
@@ -178,126 +179,233 @@
...
+                                              (if slug
+                                                (recur
+                                                  actions
+                                                  (+ col 2)
+                                                  (conj dashcards
+                                                        {:size_y 1 :size_x 2 :row 0 :col col
+                                                         :card_id model-ref
+                                                         :visualization_settings {"virtual_card" {"display" "action"}
+                                                                                  "button.label" action-name
+                                                                                  "action_slug" slug}}))
```
Comment (metamben) [inline src/metabase/api/app.clj:374]: Wouldn't something like `(mapv (fn [[slug action-name] col] ...) actions (range (- 18 (* 2 (count actions))) Long/MAX_VALUE 2))` be more readable than the `loop`?

File: test/metabase/api/app_test.clj [hunk 2]
```diff
@@ -299,6 +295,44 @@
                 {:table-ids [(data/id :venues) (data/id :venues) Integer/MAX_VALUE]
                  :app-name (str "My test app " (gensym))})))))))
 
+(deftest scaffold-model-test
```
Comment (metamben) [inline test/metabase/api/app_test.clj:298]:
```suggestion
(deftest scaffold-app-from-model-test
```

File: test/metabase/api/app_test.clj [hunk 3]
```diff
@@ -299,6 +295,44 @@
...
+                                               {:visualization_settings {:action_slug "list-action"}}]}
+                              list-page))
+                (is (partial= {:ordered_cards [{:parameter_mappings
+                                                [{:target [:dimension [:field (mt/id :categories :id) nil]]}]}
+                                               {}
+                                               {:visualization_settings {:action_slug "update"}}
+                                               {:visualization_settings {:action_slug "delete"}}
+                                               {:visualization_settings {:action_slug "detail-action"}}]}
+                              detail-page))))))))))
+
```
Comment (metamben) [inline test/metabase/api/app_test.clj:335]: Would a test scaffolding from a mix of tables and models be interesting?

**Review by metamben — APPROVED** [review]
LGTM

File: src/metabase/api/app.clj [hunk 4]
```diff
@@ -178,126 +179,232 @@
...
-                                    :display "table"}
-                           "list" {:name (i18n/tru "List")
-                                   :display "list"}
-                           "detail" {:name (i18n/tru "Detail")
-                                     :display "object"}}]
+  [app-name source-table-ids]
+  (let [table-ids (filter number? (distinct source-table-ids))
+        extract-model-id #(if-let [[_ card-id] (and (string? %) (re-find #"^card__(\d+)$" %))]
+                           (parse-long card-id)
+                           %)
```
[the remark this answers is not in the document]
Comment (snoe) [reply src/metabase/api/app.clj]: I'd rather have this addressed in another PR

**Pull request merged**